nexus/db-model/src/schema_versions.rs

```diff
@@ -0,0 +1,798 @@
+        KnownVersion::legacy(42, 0),
+        KnownVersion::legacy(41, 0),
+        KnownVersion::legacy(40, 0),
+        KnownVersion::legacy(39, 0),
+        KnownVersion::legacy(38, 0),
+        KnownVersion::legacy(37, 1),
+        KnownVersion::legacy(37, 0),
+        KnownVersion::legacy(36, 0),
+        KnownVersion::legacy(35, 0),
+        KnownVersion::legacy(34, 0),
+        KnownVersion::legacy(33, 1),
+        KnownVersion::legacy(33, 0),
+        KnownVersion::legacy(32, 0),
+        KnownVersion::legacy(31, 0),
+        KnownVersion::legacy(30, 0),
+        KnownVersion::legacy(29, 0),
+        KnownVersion::legacy(28, 0),
+        KnownVersion::legacy(27, 0),
+        KnownVersion::legacy(26, 0),
+        KnownVersion::legacy(25, 0),
+        KnownVersion::legacy(24, 0),
+        KnownVersion::legacy(23, 1),
+        KnownVersion::legacy(23, 0),
+        KnownVersion::legacy(22, 0),
+        KnownVersion::legacy(21, 0),
+        KnownVersion::legacy(20, 0),
+        KnownVersion::legacy(19, 0),
+        KnownVersion::legacy(18, 0),
+        KnownVersion::legacy(17, 0),
+        KnownVersion::legacy(16, 0),
+        KnownVersion::legacy(15, 0),
+        KnownVersion::legacy(14, 0),
+        KnownVersion::legacy(13, 0),
+        KnownVersion::legacy(12, 0),
+        KnownVersion::legacy(11, 0),
+        KnownVersion::legacy(10, 0),
+        KnownVersion::legacy(9, 0),
+        KnownVersion::legacy(8, 0),
+        KnownVersion::legacy(7, 0),
+        KnownVersion::legacy(6, 0),
+        KnownVersion::legacy(5, 0),
+        KnownVersion::legacy(4, 0),
+        KnownVersion::legacy(3, 3),
+        KnownVersion::legacy(3, 2),
+        KnownVersion::legacy(3, 1),
+        KnownVersion::legacy(3, 0),
+        KnownVersion::legacy(2, 0),
+        KnownVersion::legacy(1, 0),
+    ]
+});
+
+/// The earliest supported schema version.
+pub const EARLIEST_SUPPORTED_VERSION: SemverVersion =
+    SemverVersion::new(1, 0, 0);
+
+/// Describes one version of the database schema
+#[derive(Debug, Clone)]
+struct KnownVersion {
+    /// All versions have an associated SemVer.  We only use the major number.
+    semver: SemverVersion,
+
+    /// Path relative to the root of the schema ("schema/crdb" in the root of
+    /// this repo) where this version's update SQL files are stored
+    relative_path: String,
+}
+
+impl KnownVersion {
+    /// Generate a `KnownVersion` for a new schema version
+    ///
+    /// `major` should be the next available integer (one more than the previous
+    /// version's major number).
+    ///
+    /// `relative_path` is the path relative to "schema/crdb" (from the root of
+    /// this repository) where the SQL files live that will update the schema
+    /// from the previous version to this version.
+    fn new(major: u64, relative_path: &str) -> KnownVersion {
+        let semver = SemverVersion::new(major, 0, 0);
+        KnownVersion { semver, relative_path: relative_path.to_owned() }
```

Comment:
I'm tempted to propose that we make the relative path for these `{semver}-{name}` so that the directory also remains sorted. But I don't feel super strongly about this.

Reply:
I thought about this but it makes the instructions for merging more complicated (you have to `git mv` your directory as part of resolving a conflict, and that wouldn't necessarily be obvious) and I'm not sure it helps that much because they don't appear in sorted order in `ls(1)` output and on GitHub anyway (because it's a lexicographic sort, not a semver-aware sort).